benchmarks/benchmark_fuzzy_join_minhash.py

```diff
@@ -0,0 +1,433 @@
+
+import seaborn as sns
+from argparse import ArgumentParser
+
+from fuzzy_join_benchmark import fetch_data, evaluate
+from utils import monitor, parse_func_repr, find_result, default_parser
+
+
+# Benchmarking accuracy and speed on actual datasets
+
+benchmark_name = "fuzzy_join_encoder_benchmark"
+
+
+@monitor(
+    memory=True,
+    time=True,
+    parametrize={
+        "encoder": ["hash", "count"],
+        "dataset_name": [
+            "Country",
+            "BasketballTeam",
+            "Drug",
+            "Device",
+            "ArtificialSatellite",
+            "Amphibian",
+            "Song",
+            "HistoricBuilding",
+            "Wrestler",
+            "EthnicGroup",
+        ],
+        "analyser": ["char_wb", "char", "word"],
+        "ngram_range": [(2, 4), (2, 3), (2, 2)],
+    },
+    save_as=benchmark_name,
+    repeat=10,
+)
+def benchmark(
+    encoder: str,
+    dataset_name: str,
+    analyser: str,
+    ngram_range: tuple,
+):
+    left_table, right_table, gt = fetch_data(dataset_name)
+
+    start_time = perf_counter()
+    joined_fj = fuzzy_join(
+        left_table,
+        right_table,
+        "left",
+        "title",
+        "title",
+        encoder=encoder,
+        analyzer=analyser,
+        ngram_range=ngram_range,
+    )
+    end_time = perf_counter()
+
+    pr, re, f1 = evaluate(
+        list(zip(joined_fj["title_x"], joined_fj["title_y"])),
+        list(zip(gt["title_l"], gt["title_r"])),
+    )
+
+    res_dic = {
+        "precision": pr,
+        "recall": re,
+        "f1": f1,
+        "time_fj": end_time - start_time,
+    }
+
+    return res_dic
+
+
+def plot(res: pd.DataFrame):
+    sns.set_theme(style="ticks", palette="pastel")
+
+    rows = []
+    for i, ser in res.iterrows():
+        times = eval(str(ser["time"]))
+        memories = eval(str(ser["memory"]))
+        precisions = eval(str(ser["precision"]))
+        recalls = eval(str(ser["recall"]))
+        f1s = eval(str(ser["f1"]))
+        time_fjs = eval(str(ser["time_fj"]))
+
+        _, _, kwargs = parse_func_repr(ser["call"])
+        for time, memory, precision, recall, f1, time_fj in zip(
+            times, memories, precisions, recalls, f1s, time_fjs
+        ):
+            rows.append(
+                (
+                    kwargs["encoder"],
+                    kwargs["analyser"],
+                    kwargs["ngram_range"],
+                    kwargs["dataset_name"],
+                    time,
+                    memory,
+                    precision,
+                    recall,
+                    f1,
+                    time_fj,
+                )
+            )
+
+    df = pd.DataFrame(
+        rows,
+        columns=[
+            "encoder",
+            "analyser",
+            "ngram_range",
+            "dataset_name",
+            "time",
+            "memory",
+            "precision",
+            "recall",
+            "f1",
+            "time_fj",
+        ],
+    )
```

Comment:
Now this is some annoying code! :joy:
Having to write that looks to me like `monitor` needs to be improved.